qcodes/instrument/base.py

```diff
@@ -401,6 +410,94 @@ def validate_status(self, verbose: bool = False) -> None:
                     print(f'validate_status: param {k}: {value}')
                 p.validate(value)
 
+    def _add_params_from_decorated_methods(self):
+
+        def special_kwargs_to_meth(self, kwarg_value):
+            """
+            Returns the attribute that has the name `kwarg_value.attr_name`
+            of this instance if the kwarg_value is of type `InstanceAttr`.
+            """
+            if isinstance(kwarg_value, InstanceAttr):
+                return getattr(self, kwarg_value.attr_name)
+            return kwarg_value
+
+        # decorated methods are required to have a specific prefix
+        has_parameter_prefix = lambda s: s.startswith(DECORATED_METHOD_PREFIX)
+
+        for obj_name in filter(has_parameter_prefix, dir(self)):
+            meth = getattr(self, obj_name)
+            # the `@add_parameter` decorator adds an attribute we check for here
+            if hasattr(meth, ADD_PARAMETER_ATTR_NAME):
+                kwargs = {
+                    key: special_kwargs_to_meth(self, val.default)
+                    for key, val in sorted(inspect.signature(meth).parameters.items())
```

Comment:
why is sorting done here?
```suggestion
                    for key, val in inspect.signature(meth).parameters.items()
```

Reply:
Artifact from my previous prototyping. Thanks for spotting :)